### PR TITLE
Evaluate script: review of command-line arguments

### PR DESCRIPTION
**Pull request by sfmig**
This PR changes the CLI arguments of the evaluate script to better match the training script.

The most common use would be to run: 
```
evaluate-detector ---trained_model_path <path-to-ckpt>
```
This will load the trained model and use its config to run the evaluation. Note that it is important that we use the config used in training, to make the same dataset and use the correct test split.

However, if a dataset or config file is specified via CLI, the parameters will be overwritten.

**Other additions**
- organises config .yaml parameters into sections (feel free to give feedback, not sure what is most intuitive).
- adds evaluate logs to MLFlow,
- adds a small fix to save frames from test set,
- adds an option to save frames in a selected output directory (if none, uses a timestamped directory),
- fixes test for saving frames and adapt to check timestamped directory.

I will do the tests in a separate PR 😬  ---> #199

### CODE REVIEW

File: crabs/detection_tracking/evaluate_model.py [hunk 1]
```diff
@@ -1,98 +1,188 @@
...
+    def get_mlflow_parameters_from_ckpt(self):
+        """Get MLflow client from ckpt path and associated params."""
+        import mlflow
 
-        Includes logging metadata about the job (CLI arguments and SLURM job IDs).
-        """
-        # Assign run name
-        self.set_run_name()
+        # roughly assert the format of the path
+        assert Path(self.trained_model_path).parent.stem == "checkpoints"
```
Comment (nikk-nikaznan) [inline crabs/detection_tracking/evaluate_model.py:70]:
```suggestion
        try:
            assert Path(self.trained_model_path).parent.stem == "checkpoints", "The model path parent directory should be 'checkpoints'"
        except AssertionError as e:
            print(f"Assertion failed: {e}")
            sys.exit(1)
```

**Review by nikk-nikaznan — APPROVED** [review]
Looking good to me. Just wonder if we should move some of the functions like the `get_..` to `evaluate_utils` to make the `evaluate_model` a bit cleaner.

File: crabs/detection_tracking/evaluate_model.py [hunk 2]
```diff
@@ -1,98 +1,188 @@
...
-        self.run_name = set_mlflow_run_name()
+        logging.info("Dataset")
+        logging.info(f"Images directories: {self.images_dirs}")
+        logging.info(f"Annotation files: {self.annotation_files}")
+        logging.info(f"Seed: {self.seed_n}")
 
-    def setup_logger(self) -> MLFlowLogger:
-        """
-        Setup MLflow logger for testing.
+    def get_mlflow_parameters_from_ckpt(self):
```
[the remark this answers is not in the document]
Comment (sfmig) [reply crabs/detection_tracking/evaluate_model.py]: moved, thanks for suggesting! ✨
